app/api/api_v1/schemas/search.py

```diff
@@ -77,34 +75,45 @@ class SearchRequestBody(CprSdkSearchParameters):
     "sources", "countries", "regions", "categories", "languages"
     """
 
-    offset: int = 0
+    offset: int = Field(
+        default=0,
+        ge=0,
+        le=500,
+    )
     """
     Where to start from in the number of query result that was 
     retrieved from the search database.
     """
-    limit: int = Field(alias="page_size", default=10)
-
-    _page_size: int = PrivateAttr(default=10)
-
-    @model_validator(mode="after")
-    def backend_limit_handling(self):
-        """
-        Backend specific requirements for limit values
-
-        This caps moth the passage and family limits to the backend limit, as well as
-        differentiating between Vespas limit and the backends limit:
 
-        For vespa the limit is the size per group result
-        For the backend this is essentially a page within that
-        """
+    page_size: int = Field(
+        default=10,
+        ge=0,
+        le=500,
+    )
 
-        self.max_hits_per_family = min(
-            self.max_hits_per_family, VESPA_SEARCH_MATCHES_PER_DOC
-        )
-        self.limit = min(self.limit, VESPA_SEARCH_LIMIT)
-        self._page_size = self.limit
-        self.limit = VESPA_SEARCH_LIMIT
-        return self
+    @field_validator("offset", mode="after")
+    @classmethod
+    def offset_to_limit(cls, offset: int, info: ValidationInfo):
```

Comment:
Do you reckon the naming of this method is clear as to what its function is?

Reply:
Sounds good! Will reword